chore(frontend): Fix all typescript errors

mariusandra opened the pull request:
## Problem

I want to create a few npm packages to export parts of the posthog app, like `@posthog/apps-common` or `@posthog/lemons` or `@posthog/a-single-insight`. The last one has been asked by customers.

However, code with errors is hard to bundle. 

## Changes

This PR fixes the remaining TypeScript errors in our codebase and removes the "ignore `*Type.ts` files when checking for errors" exception.

![2022-06-01 10 58 28](https://user-images.githubusercontent.com/53387/171367752-ae2470f1-6936-4314-b210-11cb39915670.gif)

After this, `bin/check-typescript-strict` returns 0 errors. If typegen still says something like "Found 2 errors", then you have some old `*Type.ts` files weren't deleted when the `logic.ts` file itself was deleted.

I fixed this by running `yarn typegen:clean`. This however made changes to another 61 files (added `type` to `import type`), which I'm not including in this PR, but in a [followup](https://github.com/PostHog/posthog/pull/10093).

## How did you test this code?

Fixed the errors one by one, saw the nr of errors go down.

mariusandra committed:
fix dayjs

mariusandra committed:
fix timeouts (we're not strictly speaking running in nodejs)

mariusandra committed:
export unexported type

mariusandra committed:
consolidate on a single FormInstance

mariusandra committed:
no need to rename

mariusandra committed:
fuse

mariusandra committed:
forminstance 2

mariusandra committed:
locationChanged

mariusandra committed:
BuiltLogic

mariusandra committed:
remove Type.ts exception

mariusandra committed:
fix duh

Twixes committed:
Merge branch 'master' into fix-typescript-errors

Twixes commented on bin/check-typescript-strict:
```diff
@@ -3,21 +3,17 @@
 echo "Refreshing Kea logic types and running the TypeScript compiler in dry mode..."
 echo
 
-IGNORED_SPECS=('Type.ts') # *Type.ts* files are ignored, as they are generally created by kea-typegen
```
Considering that really the only purpose of this script was making it possible to ignore certain errors, can we now just remove it and update the `typescript:check` script to be `yarn typegen:write && tsc --strict`? Would make the workflow a bit simpler to grasp

mariusandra committed:
lay off the bin/check-typescript/strict script

mariusandra committed:
Merge branch 'fix-typescript-errors' of github.com:PostHog/posthog into fix-typescript-errors

mariusandra committed:
Merge branch 'master' into fix-typescript-errors

mariusandra committed:
don't think this is ever used or useful

mariusandra committed:
no real need to hide the output

mariusandra committed:
make typescript:check do what the name says

mariusandra committed:
we're already strict

mariusandra commented:
CI errors fixed.

![image](https://user-images.githubusercontent.com/53387/171579896-f96c3183-3776-464b-99b3-14ee6a569a05.png)

... and added a small update to `yarn typescript:check`:
- added `--noEmit`, so if someone ever changes things and we do emit in `tsconfig.json`, nothing will be emitted when just checking
- added `"echo 'hey, actual output that means all worked'"` as well

mariusandra commented:
... except for whatever the `$favourite_damned_place` this is:

![image](https://user-images.githubusercontent.com/53387/171580267-0a0e9ca4-3054-4e0c-865c-d10e02f6bce1.png)

mariusandra committed:
Merge branch 'master' into fix-typescript-errors

Twixes committed:
Merge branch 'master' into fix-typescript-errors

Twixes reviewed (approved):
🦾
